src/core/qgstiledownloadmanager.cpp

```diff
@@ -124,7 +124,17 @@ void QgsTileDownloadManagerReplyWorkerObject::replyFinished()
       data = reply->readAll();
   }
 
-  emit finished( data, reply->error(), reply->errorString() );
+  QMap<QNetworkRequest::Attribute, QVariant> attributes;
+  attributes.insert( QNetworkRequest::SourceIsFromCacheAttribute, reply->attribute( QNetworkRequest::SourceIsFromCacheAttribute ) );
+  attributes.insert( QNetworkRequest::SourceIsFromCacheAttribute, reply->attribute( QNetworkRequest::RedirectionTargetAttribute ) );
+  attributes.insert( QNetworkRequest::SourceIsFromCacheAttribute, reply->attribute( QNetworkRequest::HttpStatusCodeAttribute ) );
+  attributes.insert( QNetworkRequest::SourceIsFromCacheAttribute, reply->attribute( QNetworkRequest::HttpReasonPhraseAttribute ) );
+  attributes.insert( QNetworkRequest::SourceIsFromCacheAttribute, reply->attribute( QNetworkRequest::HttpReasonPhraseAttribute ) );
```

Comment:
this does not seem right, or am I missing something?

Reply:
@wonder-sk , indeed there's a duplicate here.

Reply:
Not just the duplicate, but also `SourceIsFromCacheAttribute` is inserted multiple times...

It's a pity there doesn't seem to be a way to get all attributes and one needs to go one by one.